core/chain_makers.go

```diff
@@ -141,7 +142,9 @@ func (b *BlockGen) OffsetTime(seconds int64) {
 	if b.header.Time.Cmp(b.parent.Header().Time) <= 0 {
 		panic("block time out of range")
 	}
-	b.header.Difficulty = ethash.CalcDifficulty(b.config, b.header.Time.Uint64(), b.parent.Header())
+	if pow, ok := b.engine.(consensus.PoW); ok {
+		b.header.Difficulty = pow.CalcDifficulty(b.config, b.header.Time.Uint64(), b.parent.Header())
+	}
```

Comment:
Take care, the difficulty should work for all consensus engines, not just for PoW based ones. Clique has its own notion of difficulty too.